pkg/controllers/nodeclass/status/k8sversion.go

```diff
@@ -0,0 +1,102 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"knative.dev/pkg/logging"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	azurecache "github.com/Azure/karpenter-provider-azure/pkg/cache"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily"
+	"github.com/awslabs/operatorpkg/reasonable"
+	"github.com/blang/semver/v4"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/karpenter/pkg/utils/pretty"
+)
+
+type K8sVersionReconciler struct {
+	k8sVersionProvider imagefamily.K8sVersionProvider
+	cm                 *pretty.ChangeMonitor
+}
+
+func NewK8sVersionReconciler(provider imagefamily.K8sVersionProvider) *K8sVersionReconciler {
+	return &K8sVersionReconciler{
+		k8sVersionProvider: provider,
+		cm:                 pretty.NewChangeMonitor(),
+	}
+}
+
+func (r *K8sVersionReconciler) Register(_ context.Context, m manager.Manager) error {
+	return controllerruntime.NewControllerManagedBy(m).
+		Named("nodeclass.k8sversion").
+		For(&v1alpha2.AKSNodeClass{}).
+		WithOptions(controller.Options{
+			RateLimiter:             reasonable.RateLimiter(),
+			MaxConcurrentReconciles: 10,
```

Comment:
Looks like we do this elsewhere too but, feels like this should be either a const or configurable (per-controller? Or a single global config?)

Reply:
Adding a comment to determine why this magic number, and maybe consolidate all into one const, if we have good reason for using this value. @tallaxes do you have more context on choosing this number?

Reply:
Added a TODO comment to all usages of `MaxConcurrentReconciles: 10`.

Reply:
Add tech debt issue for tracking?

Reply:
Sounds good. Leaving as is for this PR.